### PR TITLE
Move Screen Space Shadow to multicompile and remove config requirement for DXR

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
This PR is the final PR that allow to remove the requirement of shaderconfig to use DXR.
It move the usage of screen space shadow as a multicompile instead of a define.
#pragma multi_compile SCREEN_SPACE_SHADOWS_OFF SCREEN_SPACE_SHADOWS_ON
The extra multicompile is stripped based on the configuration. Support is also added to shader graph version.

The PR remove the macro:
ENABLE_RAYTRACING, SHADEROPTIONS_RAYTRACING, 
All the relevant entry related to this change are removed from the wizard

The PR also re-introduce a small optimization for colored shadow

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/DeferredTile.shader [hunk 1]
```diff
@@ -305,6 +305,7 @@ Shader "Hidden/HDRP/DeferredTile"
             #pragma multi_compile _ OUTPUT_SPLIT_LIGHTING
             #pragma multi_compile _ DEBUG_DISPLAY
             #pragma multi_compile _ SHADOWS_SHADOWMASK /// Variant with and without shadowmask
+            #pragma multi_compile SCREEN_SPACE_SHADOWS_OFF SCREEN_SPACE_SHADOWS_ON
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/DeferredTile.shader:308]: Shader

**Comment by FrancescoC-unity** [comment]
Noticed it's all merged, but isn't it missing stripping?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/DeferredTile.shader (hunk 1 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/DeferredTile.shader:308]: For the stripper. If you don't give a name to the "implicit" version, i.e
#pragma _ SCREEN_SPACE_SHADOWS
 then you can't reject the implcit one. Here we want to reject one or the other, they are exclusive, 

this is different to have always the implicit and deciding to have the second or not like for debug
#pragma multi_compile _ DEBUG_DISPLAY

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl [hunk 2]
```diff
@@ -421,7 +421,7 @@ SHADOW_TYPE EvaluateShadow_Punctual(LightLoopContext lightLoopContext, PositionI
                                     LightData light, BuiltinData builtinData, float3 N, float3 L, float4 distances)
 {
 #ifndef LIGHT_EVALUATION_NO_SHADOWS
-    SHADOW_TYPE shadow  = 1.0;
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl:424]: how did this happen?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl:424]: In previous PR I have made the change. It was actually the right thing to do, but after discussing with Francesco, I was willing to come back to a single float even if SHADOW_TYPE  is float3 to save ALU as long as we don't have supported of colored point light shadow.

**Pull request merged**